### PR TITLE
List proc data in search

### PR DESCRIPTION
**Pull request by squirrelo**
This adds the processed data listings to the study listings so you can select by processed data when creating an analysis. You can look at this by going to the study listing page and expanding rows using the end cell arrow for each row.

~~This requires the previous pull request (#1038) to be merged first.~~ Merged.

### CODE REVIEW

**Comment by squirrelo** [comment]
Note: most of the changes seen here are because of the merge in of #1038. Only  qiita_pet/handlers/study_handlers/listing_handlers.py and  qiita_pet/handlers/templates/study_list.html have changed dramatically in the pull request.

**Comment by squirrelo** [comment]
Ready for review.

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 1]
```diff
@@ -42,13 +43,40 @@ def _get_shared_links_for_study(study):
...
+    ----------
+    user : User object
+        logged in user
+    study_proc : dict of lists
+        Dictionary keyed on study_id that lists all processed data associated
+        with that study.
+    proc_samples : dict of lists
+        Dictionary keyed on proc_data_id that lists all samples associated with
+        that processed data.
+
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: Missing the return description.

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 2]
```diff
@@ -42,13 +43,40 @@ def _get_shared_links_for_study(study):
...
-def _build_study_info(user, results=None):
-    """builds list of dicts for studies table, with all html formatted"""
+def _build_study_info(user, study_proc=None, proc_samples=None):
+    """builds list of dicts for studies table, with all html formatted
+
+    Parameters
+    ----------
+    user : User object
+        logged in user
+    study_proc : dict of lists
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: Missing optional.

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 3]
```diff
@@ -42,13 +43,40 @@ def _get_shared_links_for_study(study):
...
+        Dictionary keyed on study_id that lists all processed data associated
+        with that study.
+    proc_samples : dict of lists
+        Dictionary keyed on proc_data_id that lists all samples associated with
+        that processed data.
+
+    Notes
+    -----
+    Both study_proc and proc_samples must be passed, or neither passed.
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: I would suggest also adding a comment on the description of each parameter: `must be provided if 'proc_samples' is provided`; or something along those lines

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 4]
```diff
@@ -42,13 +43,40 @@ def _get_shared_links_for_study(study):
...
+        raise IncompetentQiitaDeveloperError(
+            'Must pass proc_samples when study_proc given')
+    elif proc_samples is not None and study_proc is None:
+        raise IncompetentQiitaDeveloperError(
+            'Must pass study_proc when proc_samples given')
+    elif proc_samples is None:
+        build_info = True
+
     # get list of studies for table
     study_list = user.user_studies.union(
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: I know that you did not change this, but it is probably worth changing: the name of this variable is misleading. It is called `study_list` but the variable is of type `set`. Renaming it to `study_set` will be  better to avoid future development burden. The main reason is that when I was reading the code I though "Lists do not have the union/intersection method", and then I realize that the variable is a set.

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 5]
```diff
@@ -42,13 +43,40 @@ def _get_shared_links_for_study(study):
     return ", ".join(shared)
 
 
-def _build_study_info(user, results=None):
-    """builds list of dicts for studies table, with all html formatted"""
+def _build_study_info(user, study_proc=None, proc_samples=None):
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: Although the code on this function looks good, I had a hard time understanding completely what the function is doing. I think it will improve the readability and maintenance of the code if parts of this function are encapsulated in their own functions. For example, the construction of the `proc_data_info` elements can be encapsulated into their own function. This also makes it easier to test.

Also, IMOO the html introduced here is hard to read. Is it possible to move it to a global so it becomes easier to read (less problems with PEP8)...

**Comment by josenavas** [comment]
@squirrelo there is a need for title in the end cell arrow for each row. When I open the page, I did not know what that means (and the fact that is explain here does not help, as users will not see the PR message).

Something like available processed data or a title that indicates what that arrow does is useful.

Also, there is a bug at some place:
screen shot 2015-04-08 at 9 59 33 am
Traceback:
```
Traceback (most recent call last):
  File "/Users/jose/.virtualenvs/qiita/lib/python2.7/site-packages/tornado/web.py", line 1346, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/Users/jose/.virtualenvs/qiita/lib/python2.7/site-packages/tornado/web.py", line 2626, in wrapper
    return method(self, *args, **kwargs)
  File "/Users/jose/qiime_software/QiiTa/qiita_pet/handlers/study_handlers/listing_handlers.py", line 274, in get
    proc_samples=proc_samples)
  File "/Users/jose/qiime_software/QiiTa/qiita_pet/handlers/study_handlers/listing_handlers.py", line 137, in _build_study_info
    proc_info = proc_data.processing_info
  File "/Users/jose/qiime_software/QiiTa/qiita_db/data.py", line 1383, in processing_info
    dynamic_info = dict(conn_handler.execute_fetchone(sql))
TypeError: 'NoneType' object is not iterable
ERROR:tornado.access:500 GET /study/search/?&user=demo@microbio.me&sEcho=675&query=&_=1428512358577 (::1) 107.01ms
```
I'm using the database created by `test_data_studies/commands.sh`.

**Comment by squirrelo** [comment]
Looks like it can't handle when there are no studies available. Good catch.

File: qiita_pet/handlers/study_handlers/listing_handlers.py (hunk 5 above)
Comment (squirrelo) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: refactored and now renders the gnarly stuff client-side usingt he datatables module itself.

**Comment by squirrelo** [comment]
Thi is only failing due to the weird longitude test bug, so ready for review again.

**Comment by josenavas** [comment]
@squirrelo I see 18 files changed, is that correct? I just updated the `cart-branch` branch to be synced with master, so only your changes should be reflected here. If so, I'll review it.

**Comment by squirrelo** [comment]
I think it's a merge issue, I'll double check

**Comment by squirrelo** [comment]
Yup, merge issue, ready for review.

**Comment by josenavas** [comment]
Awesome! Thanks for checking! Reviewing right now :smile:

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 6]
```diff
@@ -42,14 +43,46 @@ def _get_shared_links_for_study(study):
...
+    Parameters
+    ----------
+    user : User object
+        logged in user
+    study_proc : dict of lists, optional
+        Dictionary keyed on study_id that lists all processed data associated
+        with that study. Required if proc_samples given.
+    proc_samples : dict of lists, optional
+        Dictionary keyed on proc_data_id that lists all samples associated with
+        that processed data. Rquired if study_proc given.
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: `Rquired` -> `Required`

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 7]
```diff
@@ -42,14 +43,46 @@ def _get_shared_links_for_study(study):
...
+
+    Returns
+    -------
+    infolist: list of dict of lists and dicts
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: o.O possible to have a format description, this is really convoluted... Something like:`[{study_id: [{foo:bar}]}]`. I think it will help to explain what is returned here...

Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: Thinking a bit more on how this looks like and reading a bit further the code, I think you will take advantage of using namedtuples. What do you think?

Comment (squirrelo) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: Possibly. How do those work when JSON serialized?

**Comment by josenavas** [comment]
@squirrelo I'm still receiving the bug on the interface.  Way to reproduce it:
```bash
# Exporting the "release" configuration
$ export MOI_CONFIG_FP=~/qiime_software/config_demo.txt 
$ export QIITA_CONFIG_FP=~/qiime_software/config_demo.txt
# This is included in the repo - and works with no errors
$ ./test_data_studies/commands.sh
$ qiita webserver start
```
Then, login to the system using the demo@microbio.me user and try to go to the study listing page.

File: qiita_pet/handlers/study_handlers/listing_handlers.py (hunk 7 above)
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: I don't know, probably worth give it a shot and try different things. The code as it is right now is really hard to follow and I don't think that is maintainable.
On the other hand, are you forced to use JSON? The reason why I'm asking is because it looks like you're using all this information in a javascript function, so you probably can just pass the python object and use the template engine to fill those values.

Comment (squirrelo) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: It has to be JSON because all this is sent over AJAX to the datatable, which is what allows the table to be dynamic for each metadata search.

Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: Ok, it looks like serializing namedtuples to json is a bit tricky http://stackoverflow.com/q/5906831/3746629.  Giving this, I think that breaking the function in multiple functions (as I suggested before) and introducing a highly detailed documentation, will help to make this code functional.

IMOO, this function should only create the list. The outer dictionary should be created by another function, and the inner dictionary by another function. This will put a logical break up in the code that will be easier to follow, as well as easier testable, since the error will be in 1 of these 3 functions rather than in a single one as it is right now.

**Comment by squirrelo** [comment]
That error is because the processed_params_sortmeRNA table is completely empty when `./test_data_studies/commands.sh` is run, so there is no processing info to pull out. This is an issue with that script, not this pull request.

**Comment by josenavas** [comment]
Oh, thanks for digging into it. Can you open an issue to fix that script? Ideally we want to use that script so new developers/users can get a functional "demo/release" database, so they can play with it.

File: qiita_pet/handlers/study_handlers/listing_handlers.py (hunk 7 above)
Comment (squirrelo) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: Done

**Comment by squirrelo** [comment]
Anyone for review?

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 8]
```diff
@@ -57,59 +126,27 @@ def _build_study_info(user, results=None):
...
+        if build_samples:
+                proc_data = study.processed_data()
+                proc_samples = {}
+                study_proc = {study.id: proc_data}
+                for pid in proc_data:
+                    proc_samples[pid] = ProcessedData(pid).samples
+
+        study_info = _build_single_study_info(study, info)
+        # Build the proc data info list for the child row in datatable
+        study_info["proc_data_info"] = _build_proc_data_info(study, study_proc,
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: It is weird that this is done here. If "proc_data_info" is part of the study info (which it is) it should be created on the `_build_single_study_info`. Just move that call to the end of that function.

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 9]
```diff
@@ -42,14 +43,82 @@ def _get_shared_links_for_study(study):
     return ", ".join(shared)
 
 
-def _build_study_info(user, results=None):
-    """builds list of dicts for studies table, with all html formatted"""
+def _build_single_study_info(study, info):
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: This auxuliary functions are missing the docstring. I think that putting the docstring is what will make the return structure more understandable, w/o the necessity to go over the code... (that's  also what I meant by breaking the functions...)

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 10]
```diff
@@ -42,14 +43,82 @@ def _get_shared_links_for_study(study):
...
+    info["num_raw_data"] = len(study.raw_data())
+    info["status"] = status
+    info["study_id"] = study.id
+    info["pi"] = study_person_linkifier((PI.email, PI.name))
+    del info["principal_investigator_id"]
+    del info["email"]
+    return info
+
+
+def _build_proc_data_info(study, study_proc, proc_samples):
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: IMOO, this function should follow the structure of the previous one. It should be `_build_single_proc_data_info` and then you call it in a list comprehension from `_build_single_study_info`. You will have then a function for a single object, and the list comprehension will be cheaper than the appends.

**Comment by josenavas** [comment]
Thanks @squirrelo ! It looks way easier to understand. However, I still think that an extra bit of documentation and making the code more pythonic is possible, but this is getting in a really good shape! Thanks!!!

File: qiita_pet/handlers/study_handlers/listing_handlers.py (hunk 8 above)
Comment (squirrelo) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: This is to make a clear separation between proc data and study info. It also makes it so I don't have to pass as many things into the single_study_info function.

File: qiita_pet/handlers/study_handlers/listing_handlers.py (hunk 9 above)
Comment (squirrelo) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: done

**Comment by squirrelo** [comment]
Ready for review again

File: qiita_pet/handlers/study_handlers/listing_handlers.py (hunk 8 above)
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: The problem is that there is no separation. The "proc_data_info" is added
to the "study_info" dictionary, which means that the proc_data_info is part
of the study_info. That is why I think it has to be added to the
"_build_sinfle_study_info", because that is info that belongs to the study.
You're just building it ad-hoc inside the other function, instead of
building it in 2 different places. My point is that if at some point we
need the "_build_single_study_info" in another part of the code, we will
need to remember to call "_build_proc_data_info" independently, which shows
a deficiency on the code structure design.

2015-04-12 16:48 GMT-07:00 Joshua Shorenstein <notifications@github.com>:

> In qiita_pet/handlers/study_handlers/listing_handlers.py
> <https://github.com/biocore/qiita/pull/1039#discussion_r28208973>:
>
> > -            "pmid": pmids,
> > -            "status": status,
> > -            "abstract": info["study_abstract"]
> > -
> > -        })
> > +        # Build the processed data info for the study if none passed
> > +        if build_samples:
> > +                proc_data = study.processed_data()
> > +                proc_samples = {}
> > +                study_proc = {study.id: proc_data}
> > +                for pid in proc_data:
> > +                    proc_samples[pid] = ProcessedData(pid).samples
> > +
> > +        study_info = _build_single_study_info(study, info)
> > +        # Build the proc data info list for the child row in datatable
> > +        study_info["proc_data_info"] = _build_proc_data_info(study, study_proc,
>
> This is to make a clear separation between proc data and study info. It
> also makes it so I don't have to pass as many things into the
> single_study_info function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/1039/files#r28208973>.
>



-- 
Jose Navas

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 11]
```diff
@@ -42,14 +43,118 @@ def _get_shared_links_for_study(study):
...
+        info["number_samples_collected"] = 0
+    info["shared"] = _get_shared_links_for_study(study)
+    info["num_raw_data"] = len(study.raw_data())
+    info["status"] = status
+    info["study_id"] = study.id
+    info["pi"] = study_person_linkifier((PI.email, PI.name))
+    del info["principal_investigator_id"]
+    del info["email"]
+    # Build the proc data info list for the child row in datatable
+    info["proc_data_info"] = _build_single_proc_data_info(
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: Thanks for moving this here!

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 12]
```diff
@@ -42,14 +43,118 @@ def _get_shared_links_for_study(study):
...
+        study, study_proc, proc_samples)
+    return info
+
+
+def _build_single_proc_data_info(study, study_proc, proc_samples):
```
Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: I just noticed that you changed the name of the function (as I proposed) but you did not make the changes on the implementation. Ideally, you create here only the dict of a single processed data. Also, the `Returns` on the numpy-doc is not accurate.

**Comment by josenavas** [comment]
Just one more comment @squirrelo

File: qiita_pet/handlers/study_handlers/listing_handlers.py (hunk 12 above)
Comment (squirrelo) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: That logic is a bit odd considering the way that study_proc and proc_samples are formatted. Making the entire list at once is more conducive to the way that the data is structured.

Comment (josenavas) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: But then this is not building the information of a single processed data. IMOO, it is not weird. When designing this kind of operations, you try to design them in a way that the smallest chunk of work is created. Also, I don't get your point. The break up that I'm proposing is the same that the one for the study. Instead of building the entire study list, you call a function for each study. Here, you call a function for each processed data. Line 79 will change to something like:
```
info["proc_data_info"] = [_build_single_proc_data_info(pd_id, proc_samples[pd_id]) for pd_id in study_proc[study.id]]
```

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 13]
```diff
@@ -42,14 +43,118 @@ def _get_shared_links_for_study(study):
...
-    """builds list of dicts for studies table, with all html formatted"""
+def _build_single_study_info(study, info, study_proc, proc_samples):
+    """Clean up and add to the study info for HTML purposes
+
+    Parameters
+    ----------
+    study : Study object
+        The study to build information for
+    info : dict
+        Information from Study.get_info
```
Comment (ElDeveloper) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: The last two arguments are missing here.

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 14]
```diff
@@ -42,14 +43,118 @@ def _get_shared_links_for_study(study):
...
+    del info["email"]
+    # Build the proc data info list for the child row in datatable
+    info["proc_data_info"] = _build_single_proc_data_info(
+        study, study_proc, proc_samples)
+    return info
+
+
+def _build_single_proc_data_info(study, study_proc, proc_samples):
+    """Build the proc data info list for the child row in datatable
+
```
Comment (ElDeveloper) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: There's an extra blank blank line here, can you remove it?

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 15]
```diff
@@ -57,59 +162,24 @@ def _build_study_info(user, results=None):
...
-            "num_raw_data": len(study.raw_data()),
-            "pi": PI,
-            "pmid": pmids,
-            "status": status,
-            "abstract": info["study_abstract"]
-
-        })
+        # Build the processed data info for the study if none passed
+        if build_samples:
+                proc_data = study.processed_data()
```
Comment (ElDeveloper) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: The indentation in this block has a few extra spaces.

File: qiita_pet/templates/list_studies.html [hunk 16]
```diff
@@ -1,7 +1,11 @@
 {% extends sitebase.html %}
 {% block head %}
 <link rel="stylesheet" href="/static/vendor/css/jquery.dataTables.css" type="text/css">
-
+<style>
+td.details-control {
```
Comment (ElDeveloper) [inline qiita_pet/templates/list_studies.html]: This should probably go in `style.css`.

File: qiita_pet/templates/list_studies.html [hunk 17]
```diff
@@ -39,6 +79,23 @@
                 }
             } 
         });
+    // Add event listener for opening and closing details
+    $('#studies-table tbody').on('click', 'td.details-control', function () {
+      var table = $('#studies-table').DataTable();
```
Comment (ElDeveloper) [inline qiita_pet/templates/list_studies.html]: This line is under-indented.

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 18]
```diff
@@ -42,14 +43,118 @@ def _get_shared_links_for_study(study):
...
+        proc_info['pid'] = pid
+        proc_info['data_type'] = proc_data.data_type()
+        proc_info['samples'] = sorted(proc_samples[pid])
+        proc_info['processed_date'] = str(proc_info['processed_date'])
+        proc_data_info.append(proc_info)
+    return proc_data_info
+
+
+def _build_study_info(user, study_proc=None, proc_samples=None):
+    """builds list of dicts for studies table, with all html formatted
```
Comment (ElDeveloper) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: builds -> Builds and html -> HTML

File: qiita_pet/handlers/study_handlers/listing_handlers.py [hunk 19]
```diff
@@ -42,14 +43,118 @@ def _get_shared_links_for_study(study):
...
+        study and processed data info for JSON serialiation for datatables
+        Each dict in the list is a single study, and contains the text
+
+    Notes
+    -----
+    Both study_proc and proc_samples must be passed, or neither passed.
+    """
+    build_samples = False
+    # Logic check to make sure both needed parts passed
+    if study_proc is not None and proc_samples is None:
```
Comment (ElDeveloper) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: I think there's no tests for these exception cases. Would you mind adding those?

**Comment by ElDeveloper** [comment]
Ready with a "static review", overall very minor comments, haven't had a chance to test with a working system as I blew up my "demo" environment. Does anyone have a "commands.sh" file and the data to add the EMP studies to the system? I'm getting a bunch of errors due to columns not complying to the required data-types.

**Comment by ElDeveloper** [comment]
Otherwise can someone else review this?

File: qiita_pet/templates/list_studies.html (hunk 16 above)
Comment (squirrelo) [inline qiita_pet/templates/list_studies.html]: it's specific to this page, so I've left it in header.

Comment (ElDeveloper) [inline qiita_pet/templates/list_studies.html]: Ok.

File: qiita_pet/handlers/study_handlers/listing_handlers.py (hunk 13 above)
Comment (squirrelo) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: added

File: qiita_pet/handlers/study_handlers/listing_handlers.py (hunk 19 above)
Comment (squirrelo) [inline qiita_pet/handlers/study_handlers/listing_handlers.py]: added

**Comment by squirrelo** [comment]
Suggestions added, ready to go.

**Comment by ElDeveloper** [comment]
Thanks, can someone review (and finally merge) this on their system or share the files for me to review/load this in my system? It would be a matter of going through the interface as this has already been reviewed by @josenavas and me.

**Comment by ElDeveloper** [comment]
It looks great dude! This exception came up when I tried to search based on a metadata filed in the studies page:

```python
================================================================================
Time: 2015-04-14 11:47:30.252444
Message:
ERROR:

Error running SQL query: SELECT EXISTS(SELECT * FROM qiita.processed_data WHERE processed_data_id=%s)
ARGS: ('18S',)
Error: invalid input syntax for integer: "18S"
LINE 1: ...T * FROM qiita.processed_data WHERE processed_data_id='18S')
                                                                 ^

TRACE:
Traceback (most recent call last):

  File "/Users/yoshikivazquezbaeza/.virtualenvs/qiita/lib/python2.7/site-packages/tornado/web.py", line 1141, in _when_complete
    callback()

  File "/Users/yoshikivazquezbaeza/.virtualenvs/qiita/lib/python2.7/site-packages/tornado/web.py", line 1162, in _execute_method
    self._when_complete(method(*self.path_args, **self.path_kwargs),

  File "/Users/yoshikivazquezbaeza/.virtualenvs/qiita/lib/python2.7/site-packages/tornado/web.py", line 2297, in wrapper
    return method(self, *args, **kwargs)

  File "/Users/yoshikivazquezbaeza/git_sw/qiita/qiita_pet/handlers/study_handlers/listing_handlers.py", line 299, in get
    proc_samples=proc_samples)

  File "/Users/yoshikivazquezbaeza/git_sw/qiita/qiita_pet/handlers/study_handlers/listing_handlers.py", line 180, in _build_study_info
    study, study_proc, proc_samples)

  File "/Users/yoshikivazquezbaeza/git_sw/qiita/qiita_pet/handlers/study_handlers/listing_handlers.py", line 104, in _build_single_proc_data_info
    proc_data = ProcessedData(pid)

  File "/Users/yoshikivazquezbaeza/git_sw/qiita/qiita_db/base.py", line 154, in __init__
    if not self._check_id(id_):

  File "/Users/yoshikivazquezbaeza/git_sw/qiita/qiita_db/base.py", line 140, in _check_id
    "{0}_id=%s)".format(self._table), (id_, ))[0]

  File "/Users/yoshikivazquezbaeza/git_sw/qiita/qiita_db/sql_connection.py", line 486, in execute_fetchone
    with self._sql_executor(sql, sql_args) as pgcursor:

  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()

  File "/Users/yoshikivazquezbaeza/git_sw/qiita/qiita_db/sql_connection.py", line 303, in _sql_executor
    (sql, str(sql_args), e)))

QiitaDBExecutionError: 
Error running SQL query: SELECT EXISTS(SELECT * FROM qiita.processed_data WHERE processed_data_id=%s)
ARGS: ('18S',)
Error: invalid input syntax for integer: "18S"
LINE 1: ...T * FROM qiita.processed_data WHERE processed_data_id='18S')
                                                                 ^



HTTP INFO:
<strong>body</strong>: 
<strong>files</strong>: {}
<strong>protocol</strong>: http
<strong>connection</strong>: <tornado.httpserver.HTTPConnection object at 0x10a651050>
<strong>uri</strong>: /study/search/?&user=test@foo.bar&sEcho=112&query=study_title%20includes%20%22soil%22&_=1429033642672
<strong>_start_time</strong>: 1429033649.39
<strong>headers</strong>: {'Accept-Language': 'en-us', 'Accept-Encoding': 'gzip, deflate', 'X-Requested-With': 'XMLHttpRequest', 'Connection': 'keep-alive', 'Accept': 'application/json, text/javascript, */*; q=0.01', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/601.1.27+ (KHTML, like Gecko) Version/8.0.5 Safari/600.5.17', 'Host': 'localhost:1111', 'Cookie': 'user="InRlc3RAZm9vLmJhciI=|1429033402|3efbd25c076741d3cceedf0d8c66ce026b070a6b"', 'Referer': 'http://localhost:1111/study/list/'}
<strong>host</strong>: localhost:1111
<strong>version</strong>: HTTP/1.1
<strong>arguments</strong>: {'query': ['study_title includes "soil"'], 'user': ['test@foo.bar'], '_': ['1429033642672'], 'sEcho': ['112']}
<strong>_finish_time</strong>: 1429033650.24
<strong>query</strong>: &user=test@foo.bar&sEcho=112&query=study_title%20includes%20%22soil%22&_=1429033642672
<strong>path</strong>: /study/search/
<strong>_cookies</strong>: Set-Cookie: user="InRlc3RAZm9vLmJhciI=|1429033402|3efbd25c076741d3cceedf0d8c66ce026b070a6b"
<strong>method</strong>: GET
<strong>remote_ip</strong>: ::1


Info:
```

**Comment by squirrelo** [comment]
Thanks @ElDeveloper, was a problem with dict formatting without searches. Fixed

**Comment by ElDeveloper** [comment]
Thanks @squirrelo
